File: meme_inator_front/lib/core/api/enums/feed_type.dart
/// Feed type enums for grid and sectional feeds
enum GridFeedType {
  recent('recent'),
  randomized('randomized'),
  videosOnly('videos_only'),
  imagesOnly('images_only'),
  mostCommented('most_commented'),
  myPosts('my_posts'),
  friends('friends'),
  following('following');

  final String value;
  const GridFeedType(this.value);
  
  @override
  String toString() => value;
}

enum SectionalFeedType {
  popularToday('popular-today'),
  popularWeekly('popular-weekly'),
  popularMonthly('popular-monthly');

  final String value;
  const SectionalFeedType(this.value);
  
  @override
  String toString() => value;
}

File: meme_inator_back/apps/feeds/domain/enums/feed_type.py
from enum import Enum
from typing import Dict

class FeedType(str, Enum):
    SECTIONAL = 'sectional'
    GRID = 'grid'

    
class GridFeedType(str, Enum):
    RECENT = "recent"
    RANDOMIZED = "randomized"
    VIDEOS_ONLY = "videos_only"
    IMAGES_ONLY = "images_only"
    MOST_COMMENTED = "most_commented"
    USER_PROFILE = "user_profile"
    FRIENDS_POSTS = "friends_posts"
    FOLLOWINGS_POSTS = "followings_posts"
    USER_UPVOTED_POSTS = "user_upvoted_posts"
    COMMENTED_FEEDS = "commented_feeds"

    @staticmethod
    def get_gridfeed_values() -> list[str]:
        return [item.value for item in GridFeedType]


class SectionalFeedType(str, Enum):
    POPULAR_TODAY = "popular-today"
    POPULAR_WEEKLY = "popular-weekly"
    POPULAR_MONTHLY = "popular-monthly"
    USER_PROFILE = "user_profile"


    @staticmethod
    def get_sectionalfeed_values() -> list[str]:
        return [item.value for item in SectionalFeedType]